recipe-client-addon/lib/FilterExpressions.jsm

```diff
@@ -41,3 +43,32 @@ this.FilterExpressions = {
     return jexl.eval(onelineExpr, context);
   },
 };
+
+/**
+ * Return an array of the given object's own keys (specifically, its enumerable
+ * properties), or undefined if the argument isn't an object.
+ * @param {Object} obj
+ * @return {Array[String]|undefined}
+ */
+function keys(obj) {
+  if (typeof obj !== "object" || obj === null) {
+    return undefined;
```

Comment:
Unfortunately `return` and `return undefined` behave differently for eslint's "you must either never return something or always return something" rules, which I guess is why `undefined` was made explicit here.